structure_tests/drivers/docker_driver.go

```
@@ -0,0 +1,357 @@
+	if err != nil {
+		panic(err)
+	}
+	originalImage = image
+	currentImage = image
+	return DockerDriver{
+		cli: *newCli,
+	}
+}
+
+func (d DockerDriver) ProcessCommand(t *testing.T, envVars []unversioned.EnvVar, fullCommand []string,
+	shellMode bool, checkOutput bool) (string, string, int) {
+
+	if len(fullCommand) == 0 {
+		t.Logf("empty command provided: skipping...")
+		return "", "", -1
+	}
+	var env []string
+	for _, envVar := range envVars {
+		env = append(env, envVar.Key+"="+envVar.Value)
+	}
+	stdout, stderr, exitCode := d.exec(t, env, fullCommand)
+
+	if stdout != "" {
+		t.Logf("stdout: %s", stdout)
+	}
+	if stderr != "" {
+		t.Logf("stderr: %s", stderr)
+	}
+	//reset image for next test
+	currentImage = originalImage
+	return stdout, stderr, exitCode
+}
+
+func (d DockerDriver) SetEnvVars(t *testing.T, vars []unversioned.EnvVar) []unversioned.EnvVar {
+	// currently, this does not support substitution (e.g. PATH=/env/bin:$PATH), so setting env vars
+	// is broken. need to find a way to retrieve the existing values from the container before setting
+	if len(vars) == 0 {
+		return nil
+	}
+	ctx := context.Background()
+	container, err := d.cli.CreateContainer(docker.CreateContainerOptions{
+		Config: &docker.Config{
+			Image:        currentImage,
+			Cmd:          []string{"NOOP_COMMAND_DO_NOT_RUN"},
+			AttachStdout: true,
+			AttachStderr: true,
+		},
+		HostConfig:       nil,
+		NetworkingConfig: nil,
+		Context:          ctx,
+	})
+	if err != nil {
+		t.Errorf("Error creating container: %s", err.Error())
+		return nil
+	}
+	image, err := d.cli.InspectContainer(container.ID)
+	if err != nil {
+		t.Errorf("Error when inspecting container: %s", err.Error())
+		return nil
+	}
+	// convert env to map for easier processing
+	imageEnv := make(map[string]string)
+	for _, varPair := range image.Config.Env {
+		pair := strings.Split(varPair, "=")
+		imageEnv[pair[0]] = pair[1]
+	}
+
+	before, _ := regexp.Compile(".*\\$(.*?):")
```

You can use MustCompile to avoid needing to swallow the errors.

---

done